### PR TITLE
Fix testRecentPrimaryInformation

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java [hunk 1]
```diff
@@ -133,7 +136,7 @@ public void testRecentPrimaryInformation() throws Exception {
             // AllocationService only calls GatewayAllocator if there are unassigned shards
             execute("""
                 create table doc.dummy (x int)
-                with ("routing.allocation.require.attr" = 'not-found')
+                with ("number_of_replicas" = 1)
```
Comment (seut) [inline server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java:139]: shouldn't this also contain `wait_for_active_shards=0`? at least the original test does this..

**Review by seut — APPROVED** [review]
Lgtm otherwise

File: server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java (hunk 1 above)
Comment (seut) [reply server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java:139]: But looks like this has no effect at this test and such is unrelated, so ignore my comment.

Comment (mfussenegger) [reply server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java:139]: I'll add it anyway :+1:

**Pull request merged**